ddl/generated_column.go

```diff
@@ -224,16 +224,35 @@ func checkModifyGeneratedColumn(tbl table.Table, oldCol, newCol *table.Column, n
 }
 
 type illegalFunctionChecker struct {
-	found bool
+	hasIllegalFunc bool
+	hasGroupByFunc bool
 }
 
 func (c *illegalFunctionChecker) Enter(inNode ast.Node) (outNode ast.Node, skipChildren bool) {
 	switch node := inNode.(type) {
 	case *ast.FuncCallExpr:
+		// blocked functions
 		if _, found := expression.IllegalFunctions4GeneratedColumns[node.FnName.L]; found {
-			c.found = true
+			c.hasIllegalFunc = true
 			return inNode, true
 		}
+		// if it is not a build in function, throw error
+		if _, found := expression.Funcs[node.FnName.L]; !found {
+			c.hasIllegalFunc = true
+			return inNode, true
+		}
+	case *ast.AggregateFuncExpr:
+		// Aggregate function is not allowed
+		c.hasGroupByFunc = true
+		return inNode, true
+	case *ast.ValuesExpr:
```

Comment:
```suggestion
	case *ast.ValuesExpr, *ast.UnionSelectList:
```
And remove the `*ast.UnionSelectList` arm.